### PR TITLE
[Sql] BREAKING CHANGE: Sdk major version bump

### PR DESCRIPTION
**Pull request by MilanBrkicFON**
**Description**

In this PR azure-mgmt-sql is bumped to latest version which is a new major version. A lot of changes were introduced:
1. Response for commands in `az sql mi` has breaking change where new parameters are returned (currentBackupsStorageRedundancy, requestedBackupStorageRedundancy)
2. Remove tde list-activity command as it is no longer supported. 
3. Adding support  for update backup storage redundancy for `sql mi set`

**Testing Guide**

for (3) added new test that is specifically testing this feature.

**History Notes**

[Sql] BREAKING CHANGE: `az sql db tde list-activity`: Command no longer exists
[Sql] BREAKING CHANGE: `az sql mi show/create/update/list`: Instead of `backupStorageRedundancy`, `currentBackupsStorageRedundancy` and `requestedBackupStorageRedundancy` properties are returned

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by evelyn-ys** [comment]
Seems like it fails in check `sql mi create` help args:
```
Test failed. Help args are different.

CLI help args:--collation, --public-data-endpoint-enabled, --key-id -k, --name -n, --backup-storage-redundancy --bsr, --assign-identity -i, --enable-ad-only-auth, --external-admin-principal-type, --location -l, --proxy-override, --user-assigned-identity-id -a, --external-admin-name, --resource-group -g, --identity-type -t, --minimal-tls-version, --vnet-name, --pid --primary-user-assigned-identity-id, --yes -y, --license-type, --edition --tier -e, --external-admin-sid, --tags, --capacity -c, --family -f, --subnet, --only-show-errors, --no-wait, --timezone-id, --storage, --maint-config-id -m

Doc help args: --collation, --public-data-endpoint-enabled, --admin-password -p, --key-id -k, --backup-storage-redundancy --bsr, --name -n, --assign-identity -i, --enable-ad-only-auth, --external-admin-principal-type, --location -l, --proxy-override, --user-assigned-identity-id -a, --external-admin-name, --resource-group -g, --identity-type -t, --minimal-tls-version, --vnet-name, --pid --primary-user-assigned-identity-id, --yes -y, --license-type, --edition --tier -e, --external-admin-sid, --tags, --capacity -c, --admin-user -u, --family -f, --subnet, --only-show-errors, --no-wait, --timezone-id, --storage, --maint-config-id -m.
```

**Comment by evelyn-ys** [comment]
Seems like it fails in check `sql mi create` help args:
```
Test failed. Help args are different.

CLI help args:--collation, --public-data-endpoint-enabled, --key-id -k, --name -n, --backup-storage-redundancy --bsr, --assign-identity -i, --enable-ad-only-auth, --external-admin-principal-type, --location -l, --proxy-override, --user-assigned-identity-id -a, --external-admin-name, --resource-group -g, --identity-type -t, --minimal-tls-version, --vnet-name, --pid --primary-user-assigned-identity-id, --yes -y, --license-type, --edition --tier -e, --external-admin-sid, --tags, --capacity -c, --family -f, --subnet, --only-show-errors, --no-wait, --timezone-id, --storage, --maint-config-id -m

Doc help args: --collation, --public-data-endpoint-enabled, --admin-password -p, --key-id -k, --backup-storage-redundancy --bsr, --name -n, --assign-identity -i, --enable-ad-only-auth, --external-admin-principal-type, --location -l, --proxy-override, --user-assigned-identity-id -a, --external-admin-name, --resource-group -g, --identity-type -t, --minimal-tls-version, --vnet-name, --pid --primary-user-assigned-identity-id, --yes -y, --license-type, --edition --tier -e, --external-admin-sid, --tags, --capacity -c, --admin-user -u, --family -f, --subnet, --only-show-errors, --no-wait, --timezone-id, --storage, --maint-config-id -m.
```

File: src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py [hunk 1]
```diff
@@ -992,6 +996,7 @@ def test_sql_db_short_term_retention(self):
 
 
 class SqlServerDbLongTermRetentionScenarioTest(ScenarioTest):
+    @live_only()
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py:999]: Why this test can't run in record mode?

File: src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py [hunk 2]
```diff
@@ -3881,6 +3877,8 @@ def test_sql_zone_resilient_pool(self, resource_group, resource_group_location,
                      JMESPathCheck('dtu', 250),
                      JMESPathCheck('zoneRedundant', True)])
 
+    # Skipping test because provisioning of servers are not available on euap region, but feature is only on euap region
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py:3880]: How can feature be available in euap region without server available?

Comment (MilanBrkicFON) [reply src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py]: changed api version in yaml files per you suggestion

File: src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py (hunk 1 above)
Comment (MilanBrkicFON) [reply src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py]: recorded

**Review by evelyn-ys — APPROVED** [review]

**Comment by jiasli** [comment]
This PR breaks CI:

https://dev.azure.com/azure-sdk/public/_build/results?buildId=1466412&view=logs&j=ad51cf76-b294-5bde-777e-be77cc5f3050&t=0e02ab23-4098-514d-f7b9-ab422251bbb2

```
2022-03-29T02:16:29.3378803Z         except CannotOverwriteExistingCassetteException as ex:
2022-03-29T02:16:29.3379238Z >           raise AssertionError(ex)
2022-03-29T02:16:29.3380350Z E           AssertionError: Can't overwrite existing cassette ('/opt/az/lib/python3.8/site-packages/azure/cli/command_modules/monitor/tests/latest/recordings/test_metric_alert_for_sql_database_scope.yaml') in your current record mode ('once').
2022-03-29T02:16:29.3382083Z E           No match for the request (<Request (PUT) https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/cli_test_metric_alert_v1_2000001/providers/Microsoft.Sql/servers/clitestservermatricalertA000002/databases/cliautomationdb01?api-version=2021-05-01-preview>) was found.
2022-03-29T02:16:29.3383114Z E           Found 1 similar requests with 1 different matcher(s) :
2022-03-29T02:16:29.3383519Z E           
2022-03-29T02:16:29.3384713Z E           1 - (<Request (PUT) https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/cli_test_metric_alert_v1_2000001/providers/Microsoft.Sql/servers/clitestservermatricalertA000002/databases/cliautomationdb01?api-version=2021-02-01-preview>).
2022-03-29T02:16:29.3385863Z E           Matchers succeeded : ['method', 'scheme', 'host', 'port', 'path']
2022-03-29T02:16:29.3386340Z E           Matchers failed :
2022-03-29T02:16:29.3386940Z E           _custom_request_query_matcher - assertion failure :
2022-03-29T02:16:29.3387367Z E           None
2022-03-29T02:16:29.3387562Z 
2022-03-29T02:16:29.3388186Z /opt/az/lib/python3.8/site-packages/azure/cli/testsdk/base.py:299: AssertionError
2022-03-29T02:16:29.3389118Z ----------------------------- Captured stdout call -----------------------------
2022-03-29T02:16:29.3389583Z None
2022-03-29T02:16:29.3390202Z ------------ generated xml file: /azure_cli_test_result/monitor.xml ------------
2022-03-29T02:16:29.3390794Z =========================== short test summary info ============================
2022-03-29T02:16:29.3391462Z FAILED tests/latest/test_monitor_metric_alert_scenarios.py::MonitorTests::test_metric_alert_for_sql_database_scope
```

**Pull request merged**